onnxscript/optimizer/_constant_folding.py

```diff
@@ -914,6 +916,22 @@ def merge_dims(dim1, dim2):
     return ir.Shape([merge_dims(dim1, dim2) for dim1, dim2 in zip(shape1, shape2)])
 
 
+def _record_contributing_values(original_node: ir.Node, replacement: Replacement) -> None:
+    """Record the set of original input values that contributed to the constant-folded outputs."""
+    folded_from: set[str] = set()
+    for input in original_node.inputs:
+        if input is None:
+            continue
+        folded_from.union(input.meta.get(_FOLDED_FROM_KEY, set()))
+        assert input.name is not None
+        folded_from.add(input.name)
+
+    for new_output in replacement.new_outputs:
+        if new_output is None:
+            continue
+        new_output.meta[_FOLDED_FROM_KEY] = folded_from
```

Comment:
Does the serializer store `str(folder_from)` in the metadata? Just curious what types of values are allowed in `value.meta[...]`